Implement tidb database monitoring (#20826)
DataDog/integrations-core: mysql/README.md

Query OK, 0 rows affected (0.00 sec)
```

##### TiDB-specific setup

> For TiDB databases, the user setup is similar but with some differences:

Reviewer: ```suggestion
For TiDB databases, the user setup is similar but with some differences:
```
Can you clarify what the set up is similar to? Is it to other Databases?

Author: I'll add `similart to other databases like MySQL, MariaDB, and so on.`

To collect `extra_performance_metrics`, your MySQL server must have `performance_schema` enabled - otherwise set `extra_performance_metrics` to `false`. For more information on `performance_schema`, see [MySQL Performance Schema Quick Start][9].

> ##### TiDB Configuration

Reviewer: ```suggestion
##### TiDB configuration
```

Author: fixed in
https://github.com/DataDog/integrations-core/pull/20826/commits/48bec2a5af3bec0e801b62bbf5bb11ea4422230c

| ---------------------- | ----------- |
| mysql.info.schema.size | GAUGE       |

#### TiDB limitations

> When using this integration with TiDB, be aware of the following limitations:

Reviewer: ```suggestion
When using this integration with TiDB, be aware of the following limitations:
```
Which integration is this referencing? If it is the InnoDB integration, would this change work: 

`When using he InnoDB integration with TiDB, be aware of the following limitations:`

Author: this might not be satisfy your request. but fixed for some.

https://github.com/DataDog/integrations-core/pull/20826/commits/48bec2a5af3bec0e801b62bbf5bb11ea4422230c

In fact I don't mention for the specific integration...
thanks for the review

- **InnoDB metrics**: TiDB doesn't use the InnoDB storage engine, so all InnoDB-related metrics are unavailable
- **Performance Schema**: TiDB doesn't have MySQL's `performance_schema`, so performance metrics requiring it are unavailable
- **Replication metrics**: TiDB uses a different replication mechanism (Raft consensus), so traditional MySQL replication metrics don't apply
- **MyISAM metrics**: TiDB doesn't support MyISAM, so key cache metrics are unavailable
- **Binary log metrics**: TiDB has a different binlog implementation, so traditional MySQL binlog metrics may not be accurate
- **Statement metrics**: TiDB uses `information_schema.cluster_statements_summary` instead of `performance_schema.events_statements_summary_by_digest`
- **Activity monitoring**: TiDB uses `information_schema.cluster_processlist` instead of `performance_schema.events_statements_current`

For Database Monitoring features:
- Query samples and explain plans are collected from `cluster_statements_summary` with some approximations
- Wait events are not available as TiDB doesn't track them in the same way as MySQL

> - Some query metrics are approximated (e.g., rows examined is estimated from keys processed)

Reviewer: ```suggestion
- Some query metrics are approximated (for example, rows examined is estimated from keys processed)
```

Author: thanks, applied in
https://github.com/DataDog/integrations-core/pull/20826/commits/48bec2a5af3bec0e801b62bbf5bb11ea4422230c

- [Database user lacks privileges][29]
- [How to collect metrics with a SQL Stored Procedure?][30]

### TiDB-specific troubleshooting

> **Missing metrics**: If you see warnings about missing InnoDB or performance_schema metrics when monitoring TiDB:

Reviewer: ```suggestion
**Missing metrics**: If you see warnings about missing InnoDB or `performance_schema` metrics when monitoring TiDB:
```
